azurerm/resource_arm_relay_hybrid_connection.go

```diff
@@ -0,0 +1,192 @@
+			RequiresClientAuthorization: &requireClientAuthroization,
+			UserMetadata:                &userMetadata,
+		},
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, relayNamespace, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating Relay Hybrid Connection %q (Namespace %q Resource Group %q): %+v", name, relayNamespace, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, relayNamespace, name)
+	if err != nil {
+		return fmt.Errorf("Error issuing get request for Relay Hybrid Connection %q (Namespace %q Resource Group %q): %+v", name, relayNamespace, resourceGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Relay Hybrid Connection %q (Namespace %q Resource group %s) ID", name, relayNamespace, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmHybridConnectionRead(d, meta)
+}
+
+func resourceArmHybridConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Relay.HybridConnectionsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	relayNamespace := id.Path["namespaces"]
+	name := id.Path["hybridConnections"]
+
+	resp, err := client.Get(ctx, resourceGroup, relayNamespace, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Relay Hybrid Connection %q (Namespace %q Resource Group %q): %s", name, relayNamespace, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("relay_namespace_name", relayNamespace)
+
+	if props := resp.HybridConnectionProperties; props != nil {
+		d.Set("requires_client_authorization", props.RequiresClientAuthorization)
+		d.Set("user_metadata", props.UserMetadata)
+	}
+
+	return nil
+}
+
+func resourceArmHybridConnectionDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Relay.HybridConnectionsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	relayNamespace := id.Path["namespaces"]
+	name := id.Path["hybridConnections"]
+
+	log.Printf("[INFO] Waiting for Relay Hybrid Connection %q (Namespace %q Resource Group %q) to be deleted", name, relayNamespace, resourceGroup)
+	rc, err := client.Delete(ctx, resourceGroup, relayNamespace, name)
+
+	if err != nil {
+		if response.WasNotFound(rc.Response) {
+			return nil
+		}
+
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Pending"},
+		Target:     []string{"Deleted"},
+		Refresh:    hybridConnectionDeleteRefreshFunc(ctx, client, resourceGroup, relayNamespace, name),
+		Timeout:    60 * time.Minute,
```

Comment:
In the [Timeout](https://github.com/terraform-providers/terraform-provider-azurerm/pull/4832/files#diff-f981b0107e947bfa3d8bc26df9924a16R33) section defined in the schema, you defined the Delete timeout as 30 minutes. We should sync these up so this is either 30 minutes or the delete timeout defined above is 60 minutes.